core/arch/arm/mm/core_mmu.c

```diff
@@ -107,21 +128,22 @@ TEE_Result cache_op_outer(enum cache_op op, paddr_t pa, size_t len)
 		break;
 	case DCACHE_AREA_INVALIDATE:
 		if (len)
-			arm_cl2_invbypa(pl310_base(), pa, pa + len - 1);
+			arm_cl2_invbypa(pl310_base_pa_op, pa, pa + len - 1);
```

Comment:
Doing `DCACHE_AREA_INVALIDATE` on non-secure memory seems a bit risky. How can we be sure not to break anything in normal world?

Reply:
An invalidate operation is always risky if used carelessly. How to prevent the user from a misuse ? I'm not sure that we can prevent it.

Reply:
Why must we do this instead of using a secure bounce buffer? I assume it's for performance, but is it needed that bad? Do we have any use cases where the performance would be unacceptable?

Reply:
Some crypto operations failure happen when the CAAM is enabled on platforms with PL310 controllers because of the missing cache management for output buffers. 
The CAAM driver implements SGT management to improve performance and avoid data copy. A secure bounce buffer would impact performance and defeat the purpose of the SGT.
If a developer is not comfortable using cache invalidation, it's up to the driver to implement a secure bounce buffer. However, I think developers should have the choice between the two approaches.

Reply:
Perhaps it's possible to do both? If buffers are cache-line aligned then we can work directly with them, otherwise a secure bounce buffer is needed.

Reply:
The CAAM already implements that feature
https://github.com/OP-TEE/optee_os/blob/dc6563d7d6a4a1ab83e506aa101819a88efe418c/core/drivers/crypto/caam/utils/utils_dmaobj.c#L336-L346